packages/doenetml-worker-rust/lib-doenetml-core/src/core/general_state_var/mirror_state_var.rs

```diff
@@ -0,0 +1,71 @@
+#[cfg(test)]
+#[path = "mirror_state_var.test.rs"]
+mod tests;
+
+use crate::components::prelude::*;
+
+/// A struct of all data required to compute the value of this state variable.
+#[add_dependency_data]
+#[derive(Debug, Default, StateVariableDependencies, StateVariableDataQueries)]
+pub struct MirrorRequiredData<T>
+where
+    T: Default + Clone,
+{
+    mirrored_value: StateVarView<T>,
+}
+
+/// A state variable that mirrors another state variable from the same component.
+///
+/// The state variable has a single data query, which is for the value of another state variable.
+/// It takes on the value of this other state variable,
+/// and when inverting, requests that the value of this other variable be changed.
+#[derive(Debug, Default)]
+pub struct MirrorStateVar {
+    mirrored_state_var_idx: StateVarIdx,
+}
+
+impl MirrorStateVar {
+    pub fn new(mirrored_state_var_idx: StateVarIdx) -> Self {
+        MirrorStateVar {
+            mirrored_state_var_idx,
+        }
+    }
+}
+
+impl<T> StateVarUpdater<T, MirrorRequiredData<T>> for MirrorStateVar
+where
+    T: Default + Clone + std::fmt::Debug,
+    StateVarView<T>: TryFromState<StateVarViewEnum>,
+    <StateVarView<T> as TryFromState<StateVarViewEnum>>::Error: std::fmt::Debug,
+{
+    fn return_data_queries(&self) -> Vec<Option<DataQuery>> {
+        MirrorRequiredDataQueries {
+            mirrored_value: Some(DataQuery::StateVar {
+                component_idx: None,
+                state_var_idx: self.mirrored_state_var_idx,
```

Comment:
Is this always mirroring a state var on the same component? Or can this work cross-component?

Reply:
I hope that eventually we could make this cross-component. We just need to determine an approach for getting the `component_idx` of the desired component into the state variable updater. Since it wasn't obvious to me how this will work yet, I left it intra-component until we find a use-case where we need the cross-component feature.